arangod/Replication/ReplicationClients.cpp

```diff
@@ -155,9 +175,41 @@ void ReplicationClientsProgressTracker::garbageCollect(double thresholdStamp) {
 /// returns UINT64_MAX in case no clients are registered
 uint64_t ReplicationClientsProgressTracker::lowestServedValue() const {
   uint64_t value = UINT64_MAX;
+  if (_clients.empty()) {
+    return value;
+  }
   READ_LOCKER(readLocker, _lock);
+  decltype(_clients)::key_type minKey{};
   for (auto const& it : _clients) {
+    if (it.second.lastServedTick < value) {
+      minKey = it.first;
+    }
     value = std::min(value, it.second.lastServedTick);
   }
+  LOG_TOPIC("hunde", INFO, Logger::FIXME)
+      << "lowestServedValue(): min=" << value << " reached by syncer=" << minKey;
   return value;
 }
+
+void ReplicationClientsProgressTracker::untrack(SyncerId const syncerId,
+                                                std::string const& clientId) {
+  std::string key = getKey(syncerId, clientId);
+  auto const syncer = syncerId.toString();
+  LOG_TOPIC("69c75", TRACE, Logger::REPLICATION)
+      << "removing replication client entry for syncer " << syncer
+      << " from client " << clientId;
+  _clients.erase(key);
+}
+
+ReplicationClientsProgressTracker::~ReplicationClientsProgressTracker() {
+  if (!_clients.empty()) {
```

Comment:
do we really want to dump this info in the dtor?
should be restricted to maintainer mode IMHO

Reply:
My intention was that all inserts and removes from this could be traced, but I suppose the usefulness is quite restricted as the objects won't be destructed a lot.
I'll restrict it to maintainer mode.